### PR TITLE
add search by name field

### LINKED ISSUE #46: Search by mentor's name
When the list gets longer, it may be a good idea to search by the mentor's name

### PR DESCRIPTION
**Pull request by mohamednajiub**
Add filter mentors by name

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -111,6 +112,14 @@ function App() {
     }
   };
 
+  const searchByMentorName = (mentorName) => {
+    if(mentorName.length >= 3 ){
+      setMentors(mentorsList.filter((mentor) => mentor.name.toLowerCase().indexOf(mentorName.toLowerCase()) !== -1));
+    } else {
+      setMentors(mentorsList);
```
Comment (islamCodehood) [inline src/App.js]: Thanks @mohamednajiub  for the contribution.

You need to link this filter with the other filters: by skill, country, hearted mentors.
Please, see the function: **filterMentors** and start from there.

Comment (mohamednajiub) [reply src/App.js]: Thanks @islamCodehood  for this hint I'll fix it.

**Comment by mohamednajiub** [comment]
Hi @islamCodehood  I fixed it can you please review the PR one more time

File: src/App.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ function App() {
   const [mentors, setMentors] = useState(mentorsList);
   const [darkMode, setDarkMode] = useState(false);
 
+  const [name, setName] = useState("");
```
Comment (islamCodehood) [inline src/App.js:38]: Thanks for sticking with the state naming convention used.

File: src/App.js [hunk 3]
```diff
@@ -60,57 +61,136 @@ function App() {
     const filteredHeartedMentors = mentorsList.filter((mentor) =>
       heartedMentors.includes(mentor.id)
     );
-    if (skill === "" && country === "") {
-      setMentors(filteredHeartedMentors);
-    } else if (skill === "") {
+    // if (name === "" && skill === "" && country === "") {
```
Comment (islamCodehood) [inline src/App.js:64]: Please, don't include your commented out code lines in your PR.

File: src/App.js [hunk 4]
```diff
@@ -60,57 +61,136 @@ function App() {
...
+        .filter((mentor) => mentor.skills.includes(skill))
       );
     }
+    
   };
+
   const filterMentors = () => {
     switch(isFavMentors) {
       case true:
-        filterFromHeartedMentors(skill, country);
```
Comment (islamCodehood) [inline src/App.js:104]: Great omitting unnecessary args here :+1:

**Review by islamCodehood — APPROVED** [review]
Thanks for PR, @mohamednajiub . I am gonna merge it to [search_by_name branch](https://github.com/codehood-org/mentors-website/tree/search_by_name) and test it before merging to master.

**Pull request merged**